src/Microsoft.DotNet.CodeFormatting/Rules/MarkReadonlyFieldsRule.cs

```diff
@@ -249,6 +249,20 @@ public override void VisitAssignmentExpression(AssignmentExpressionSyntax node)
                 CheckForFieldWrite(node.Left);
             }
 
+            public override void VisitPostfixUnaryExpression(PostfixUnaryExpressionSyntax node)
+            {
+                base.VisitPostfixUnaryExpression(node);
+
+                CheckForFieldWrite(node.Operand);
+            }
+
+            public override void VisitPrefixUnaryExpression(PrefixUnaryExpressionSyntax node)
+            {
+                base.VisitPrefixUnaryExpression(node);
+
+                CheckForFieldWrite(node.Operand);
```

Comment:
I think this should check for Increment operators, because the other unary operators like ! (negation) or ~ (bitwise complement) would be ok. I would do this like so:
```C#
            public override void VisitPrefixUnaryExpression(PrefixUnaryExpressionSyntax node)
            {
                base.VisitPrefixUnaryExpression(node);

                switch (node.OperatorToken.Kind())
                {
                    case SyntaxKind.MinusMinusToken:
                    case SyntaxKind.PlusPlusToken:
                        {
                            CheckForFieldWrite(node.Operand);
                            break;
                        }
                }
            }
```